pdf-toolbox/src/main/java/com/lowagie/tools/ConcatPdf.java

```diff
@@ -77,54 +83,63 @@ public static void main(String[] args) {
         }
         else {
             try {
-                int pageOffset = 0;
-                List<Map<String, Object>> master = new ArrayList<>();
-                int f = 0;
-                String outFile = args[args.length-1];
-                Document document = null;
-                PdfCopy  writer = null;
-                while (f < args.length-1) {
-                    // we create a reader for a certain document
-                    PdfReader reader = new PdfReader(args[f]);
-                    reader.consolidateNamedDestinations();
-                    // we retrieve the total number of pages
-                    int n = reader.getNumberOfPages();
-                    List<Map<String, Object>> bookmarks = SimpleBookmark.getBookmarkList(reader);
-                    if (bookmarks != null) {
-                        if (pageOffset != 0)
-                            SimpleBookmark.shiftPageNumbersInRange(bookmarks, pageOffset, null);
-                        master.addAll(bookmarks);
-                    }
-                    pageOffset += n;
-                    System.out.println("There are " + n + " pages in " + args[f]);
-                    
-                    if (f == 0) {
-                        // step 1: creation of a document-object
-                        document = new Document(reader.getPageSizeWithRotation(1));
-                        // step 2: we create a writer that listens to the document
-                        writer = new PdfCopy(document, new FileOutputStream(outFile));
-                        // step 3: we open the document
-                        document.open();
-                    }
-                    // step 4: we add content
-                    PdfImportedPage page;
-                    for (int i = 0; i < n; ) {
-                        ++i;
-                        page = writer.getImportedPage(reader, i);
-                        writer.addPage(page);
-                        System.out.println("Processed page " + i);
-                    }
-                    writer.freeReader(reader);
-                    f++;
+                File outFile = new File(args[args.length-1]);
+                List<File> sources = new ArrayList<>();
+                for (int i = 0; i < args.length -2; i++) {
```

Comment:
I think the last document to be merged will be missing in the merged document.
```suggestion
                for (int i = 0; i < args.length -1; i++) {
```
is probably better, as i will be incremented till args.length - 2